core/tests/ts-integration/src/env.ts

```diff
@@ -43,13 +43,24 @@ export async function waitForServer(l2NodeUrl: string) {
     throw new Error('Failed to wait for the server to start');
 }
 
-function getMainWalletPk(pathToHome: string, network: string): string {
-    if (network.toLowerCase() == 'localhost') {
+function getMainWalletPk(pathToHome: string, chain?: string): string {
+    if (process.env.MASTER_WALLET_PK) {
+        return process.env.MASTER_WALLET_PK;
+    } else {
         const testConfigPath = path.join(pathToHome, `etc/test_config/constant`);
         const ethTestConfig = JSON.parse(fs.readFileSync(`${testConfigPath}/eth.json`, { encoding: 'utf-8' }));
-        return ethers.Wallet.fromPhrase(ethTestConfig.test_mnemonic).privateKey;
-    } else {
-        return ensureVariable(process.env.MASTER_WALLET_PK, 'Main wallet private key');
+
+        let wallet_name = `test_mnemonic`;
+
+        if (chain) {
+            let id = loadChainConfig(pathToHome, chain).id;
+            wallet_name += `${id + 1}`;
+        }
+
+        let pk = ethers.Wallet.fromPhrase(ethTestConfig[wallet_name]).privateKey;
+        process.env.MASTER_WALLET_PK = pk;
+
+        return pk;
```

Comment:
It's kind of a long way to go. 
You can set `MASTER_WALLET_PK` from the zk_supervisor and use the default one if env is not specified.

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/2594/commits/27d141445c95c2fcaffd5d60109e6ca79833d653